substrate/frame/support/src/storage/mod.rs

```diff
@@ -2109,6 +2149,31 @@ mod test {
 				BoundedVec::<u32, ConstU32<7>>::try_from(vec![4, 5]).unwrap(),
 			);
 		});
+
+		TestExternalities::default().execute_with(|| {
+			let bounded: BoundedVec<u32, ConstU32<7>> = vec![1, 2, 3].try_into().unwrap();
+			FooQuadMap::insert((1, 1, 1, 1), bounded);
+
+			assert_ok!(FooQuadMap::try_append((1, 1, 1, 1), 4));
+			assert_ok!(FooQuadMap::try_append((1, 1, 1, 1), 5));
+			assert_ok!(FooQuadMap::try_append((1, 1, 1, 1), 6));
+			assert_ok!(FooQuadMap::try_append((1, 1, 1, 1), 7));
+			assert_eq!(FooQuadMap::decode_len((1, 1, 1, 1)).unwrap(), 7);
+			assert!(FooQuadMap::try_append((1, 1, 1, 1), 8).is_err());
+
+			// append to a non-existing
+			assert!(FooQuadMap::get((2, 1, 1, 1)).is_none());
+			assert_ok!(FooQuadMap::try_append((2, 1, 1, 1), 4));
+			assert_eq!(
+				FooQuadMap::get((2, 1, 1, 1)).unwrap(),
+				BoundedVec::<u32, ConstU32<7>>::try_from(vec![4]).unwrap(),
+			);
+			assert_ok!(FooQuadMap::try_append((2, 1, 1, 1), 5));
+			assert_eq!(
+				FooQuadMap::get((2, 1, 1, 1)).unwrap(),
+				BoundedVec::<u32, ConstU32<7>>::try_from(vec![4, 5]).unwrap(),
```

Comment:
You can use this macro: https://paritytech.github.io/polkadot-sdk/master/frame_support/testing_prelude/macro.bounded_vec.html

Reply:
code above uses the same syntax, so i think this is fine